### PR TITLE
peazip: use system compression programs

### PR DESCRIPTION
**Pull request by auroraanna**
## Description of changes
- Updated peazip to 9.5.0.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

superseding #262788

### CODE REVIEW

File: pkgs/tools/archivers/peazip/default.nix [hunk 1]
```diff
@@ -47,36 +53,47 @@ stdenv.mkDerivation rec {
...
+      zstd
+    ]}
+    echo ${lib.makeBinPath [
+      _7z
+      archiver
+      brotli
+      upx
+      zpaq
+      zstd
+    ]}
```
Comment (wegank) [inline pkgs/tools/archivers/peazip/default.nix:80]:
```suggestion
```

**Comment by auroraanna** [comment]
Moving the binaries into PATH didn't break anything but I've been testing if all those compression algorithms and archive formats actually work

- [x] zip
- [x] brotli
- [x] zstd
- [x] xz
- [x] bzip2
- [x] gzip
- [x] iso
- [x] zpaq
- [x] wim
- [x] tar
- [] deb (somehow fails at extracting, but can open in file manager)
- [x] 7z
- [x] split

## not tested

- sfx
- ACE
- ARJ
- CAB
- CHM
- CPIO
- JAR
- LHA/LZH
- NSIS
- OOo
- PEA
- RAR (creating not tested, but extracting)
- RPM
- ZIPX

these are all listed in the *about* of [it's repository](https://github.com/peazip/PeaZip). The website says that it can handle more than 200 archive and compression formats but I couldn't find a list of all of them. For lots of these, I also don't have the option to create create new compressed files or archives.

File: pkgs/misc/air-quality-into-db/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
+{ rustPlatform
```
Comment (wegank) [inline pkgs/misc/air-quality-into-db/default.nix:1]: Looks like another unrelated file?

Comment (auroraanna) [reply pkgs/misc/air-quality-into-db/default.nix]: yes ;-;

**Comment by auroraanna** [comment]
peazip has plugins, I'm wondering if they should come with this package.

**Comment by auroraanna** [comment]
hmm, the only relevant plugin seems to be the *additional formats* one. the other 2 plugins have security issues or are now obsolete because 7zip got support for the formats they added.

the plugins are really just more executables installed into `res/bin`. for nixpkgs, all those executables would have to be packaged for in nixpkgs first.

File: pkgs/tools/archivers/peazip/default.nix [hunk 3]
```diff
@@ -16,16 +18,20 @@
...
     owner = "peazip";
     repo = pname;
     rev = version;
-    hash = "sha256-to5VhuTomw33WRWrtSIMF/SD+KVXsUKmSrJ84BNatqw=";
+    hash = "sha256-3nRptI2i5t4Ds9RisxTV/MqRXEz2aNJ4hhIBh2/9Zz8=";
   };
   sourceRoot = "${src.name}/peazip-sources";
 
+  postPatch = ''
+    substituteInPlace dev/peach.pas --replace "  HSYSBIN       = 0;" "  HSYSBIN       = 2;"
```
Comment (auroraanna) [inline pkgs/tools/archivers/peazip/default.nix:32]: This makes peazip compile so that it will use the compression and archiver executables in it's PATH.

File: pkgs/tools/archivers/peazip/default.nix [hunk 4]
```diff
@@ -47,36 +53,39 @@ stdenv.mkDerivation rec {
...
-    install -D res/share/batch/freedesktop_integration/*.desktop -t "$out/share/applications"
+    # Install desktop entries
+    install -D res/share/batch/freedesktop_integration/*.desktop -t $out/share/applications
+    # Install desktop entries's icons
+    mkdir -p $out/share/icons/hicolor/256x256/apps
+    ln -s $out/share/peazip/icons/peazip.png -t $out/share/icons/hicolor/256x256/apps/
+    mkdir $out/share/icons/hicolor/256x256/mimetypes
+    ln -s $out/share/peazip/icons/peazip_{7z,zip,cd}.png $out/share/icons/hicolor/256x256/mimetypes/
+    mkdir $out/share/icons/hicolor/256x256/actions
+    ln -s $out/share/peazip/icons/peazip_{add,extract,convert}.png $out/share/icons/hicolor/256x256/actions/
```
Comment (auroraanna) [inline pkgs/tools/archivers/peazip/default.nix:88]: No need to copy when you can link.

Comment (auroraanna) [inline pkgs/tools/archivers/peazip/default.nix:88]: This makes all the desktop entries have the their icons accessible.

**Comment by auroraanna** [comment]
putting this into https://github.com/NixOS/nixpkgs/pull/299800 now
